src/web/app/components/notification-banner/notification-banner.component.spec.ts

```diff
@@ -23,11 +59,90 @@ describe('NotificationBannerComponent', () => {
 
   beforeEach(() => {
     fixture = TestBed.createComponent(NotificationBannerComponent);
+    notificationService = TestBed.inject(NotificationService);
+    statusMessageService = TestBed.inject(StatusMessageService);
     component = fixture.componentInstance;
+    component.notificationTargetUser = NotificationTargetUser.STUDENT;
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should close after clicking X', () => {
+    jest.spyOn(notificationService, 'getUnreadNotificationsForTargetUser').mockReturnValue(of({
+      notifications: [testNotificationOne],
+    }));
+    component.ngOnInit();
+    fixture.detectChanges();
+
+    expect(component.isShown).toBeTruthy();
+    const button = fixture.debugElement.query(By.css('#btn-close-notif')).nativeElement;
+    button.click();
+    expect(component.isShown).toBeFalsy();
+  });
+
+  it('should close after clicking mark as read', () => {
+    jest.spyOn(notificationService, 'getUnreadNotificationsForTargetUser').mockReturnValue(of({
+      notifications: [testNotificationOne],
+    }));
+    jest.spyOn(notificationService, 'markNotificationAsRead').mockReturnValue(of({
+      readNotifications: [testNotificationOne.notificationId],
+    }));
+    const spy1: SpyInstance = jest.spyOn(notificationService, 'markNotificationAsRead')
+    .mockImplementation((request: MarkNotificationAsReadRequest) => {
+      expect(request.notificationId).toEqual(testNotificationOne.notificationId);
+      return of({
+        readNotifications: [request.notificationId],
+      });
+    });
+    const spy2: SpyInstance = jest.spyOn(statusMessageService, 'showSuccessToast')
```

Comment:
```suggestion
    const messageSpy: SpyInstance = jest.spyOn(statusMessageService, 'showSuccessToast')
```
Try to use more descriptive names